examples/ffi/exporter.cpp

```diff
@@ -0,0 +1,150 @@
+extern "C"
+{
+#include <ddprof/ffi.h>
+}
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
+static ddprof_ffi_ByteSlice to_byteslice(const char *s)
+{
+    return {.ptr = (uint8_t *)s,
+            .len = strlen(s)};
+}
+
+static ddprof_ffi_Slice_c_char to_slice_c_char(const char *s)
+{
+    return {.ptr = s,
+            .len = strlen(s)};
+}
```

Comment:
Not related to this PR, but this looks to me as a weird sharp edge in the libddprof API. 

It's weird to have both `ddprof_ffi_ByteSlice` and `ddprof_ffi_Slice_c_char` when they are effectively the same (?).

Also, I had to write similar helpers for using libddprof, so perhaps it would make sense to export these (possibly as macros?) in ffi.h as well :)

Reply:
You usually don't want to encourage people to transport strings without an actual size associated to it. `strlen` is also not something you want encourage, so I don't know how I feel about exposing these types of APIs. 
I guess we can also mention this :-)

Reply:
Hmmm... I'm guessing the "don't want to encourage" may be related to performance -- is that the case?

If so, would it make sense to call it something like `to_slice_slow`? I guess for a bunch of hardcoded strings ("ruby", "wall-clock", "nanoseconds", "service", ...) we just use them once so the cost is trivial so we could have the ease of use instead :)

Reply:
* I agree there is room for improvements concerning the APIs, we could provide some helpers function like `to_slice_xxx` on top of FFI headers. Maybe even a more friendly C++ interface.
* It's strange that slices for type/unit/filename/build_id/name/system_name/... are different from slices used for /tags/filenames.

But this would be subject of another pull request ;)

Reply:
I don't feel super strongly about this, but I would usually avoid relying on C style null terminated strings. Though it is not clear how we could avoid these conversions at boundaries or how to have a common types.